src/main/haskell/kore/src/Kore/Builtin/List.hs

```diff
@@ -288,3 +301,93 @@ lookupSymbolGet
     :: KoreIndexedModule attrs
     -> Either (Kore.Error e) (Kore.SymbolOrAlias Object)
 lookupSymbolGet = Builtin.lookupSymbol "LIST.get"
+
+isSymbolConcat :: MetadataTools Object Hook -> Kore.SymbolOrAlias Object -> Bool
+isSymbolConcat = Builtin.isSymbol "LIST.concat"
+
+isSymbolElement :: MetadataTools Object Hook -> Kore.SymbolOrAlias Object -> Bool
+isSymbolElement = Builtin.isSymbol "LIST.element"
+
+unify
+    :: forall level variable m p expanded proof.
+        ( OrdMetaOrObject variable, ShowMetaOrObject variable
+        , SortedVariable variable
+        , MonadCounter m
+        , MetaOrObject level
+        , Hashable variable
+        , FreshVariable variable
+        , p ~ PureMLPattern level variable
+        , expanded ~ ExpandedPattern level variable
+        , proof ~ SimplificationProof level
+        )
+    => MetadataTools level StepperAttributes
+    -> (p -> p -> Result (m (expanded, proof)))
+    -> (p -> p -> Result (m (expanded, proof)))
+unify
+    tools@MetadataTools { symbolOrAliasSorts }
+    simplifyChild
+    (DV_ dvSort (BuiltinDomainList list1))
+    (DV_ _    (BuiltinDomainList list2))
+  = give symbolOrAliasSorts $ case (list1, list2) of
+      (h1 :<| t1, h2 :<| t2) -> do
+        h12Unified <- simplifyChild h1 h2
+        let list1Tail = DV_ dvSort $ BuiltinDomainList t1
+        let list2Tail = DV_ dvSort $ BuiltinDomainList t2
+        t12Unified <- unify tools simplifyChild list1Tail list2Tail
+        return $ do
+            (h12Unified', _) <- h12Unified
+            (t12Unified', _) <- t12Unified
+            res <- 
+              normalize tools $ 
```

Comment:
Could you make your indentation a multiple of 4 throughout this entire PR, except in the cases allowed by the style guide?